src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -0,0 +1,206 @@
+                var dataRemovingVisitor = new RemoveOptionalDataVisitor(optionallyEmittedData);
+                dataRemovingVisitor.Visit(sarifLog);
+            }
+
+            optionallyEmittedData = this.FilingContext.DataToInsert;
+            if (optionallyEmittedData != OptionallyEmittedData.None)
+            {
+                var dataInsertingVisitor = new InsertOptionalDataVisitor(optionallyEmittedData);
+                dataInsertingVisitor.Visit(sarifLog);
+            }
+
+            SplittingStrategy splittingStrategy = this.FilingContext.SplittingStrategy;
+            if (splittingStrategy == SplittingStrategy.None)
+            {
+                FileWorkItemsForSarifLog(sarifLog);
+                return;
+            }
+
+            for (int runIndex = 0; runIndex < sarifLog.Runs.Count; ++runIndex)
+            {
+                if (sarifLog.Runs[runIndex]?.Results?.Count > 0)
+                {
+                    IList<SarifLog> logsToProcess = new List<SarifLog>(new SarifLog[] { sarifLog });
+
+                    if (splittingStrategy != SplittingStrategy.PerRun)
+                    {
+                        SplittingVisitor visitor;
+                        switch (splittingStrategy)
+                        {
+                            case SplittingStrategy.PerRunPerRule:
+                                visitor = new PerRunPerRuleSplittingVisitor();
+                                break;
+                            
+                            case SplittingStrategy.PerRunPerTarget:
+                                visitor = new PerRunPerTargetSplittingVisitor();
+                                break;
+                            
+                            case SplittingStrategy.PerRunPerTargetPerRule:
+                                visitor = new PerRunPerTargetPerRuleSplittingVisitor();
+                                break;
+
+                            // TODO: Implement PerResult and PerRun splittings strategies
+                            //
+                            //       https://github.com/microsoft/sarif-sdk/issues/1763
+                            //       https://github.com/microsoft/sarif-sdk/issues/1762
+                            //
+                            case SplittingStrategy.PerResult:
+                            case SplittingStrategy.PerRun:                           
+                            default:
+                                throw new ArgumentOutOfRangeException($"GroupingStrategy: {splittingStrategy}");
+                        }
+
+                        visitor.VisitRun(sarifLog.Runs[runIndex]);
+                        logsToProcess = visitor.SplitSarifLogs;
+                    }
+
+                    for (int splitFileIndex = 0; splitFileIndex < logsToProcess.Count; splitFileIndex++)
+                    {
+                        SarifLog splitLog = logsToProcess[splitFileIndex];
+                        FileWorkItemsForSarifLog(sarifLog);
```

Comment:
>FileWorkItemsForSarifLog [](start = 24, length = 24)

`FileWorkItemsForSarifLog` -- I understand you're reaching for names here, since `FileWorkItemsForLog` is already taken. But "log" is already a "SARIF log", so it's not clear to the reader how "FileWorkItemsForSarifLog" is different from "FileWorkItemsForLog".

How about `FileWorkItemsForSplitLog`.

Reply:
This is a private helper. Renamed it FileWorkItemHelper. Also made it internal and static, as this should be a primary knob for driving unit tests.

---
In reply to: [381480562](https://github.com/microsoft/sarif-sdk/pull/1775#discussion_r381480562) [](ancestors = 381480562)